src/rust/plugin-execution-sidecar/src/plugin_executor.rs

```diff
@@ -0,0 +1,69 @@
+use std::time::Duration;
```

Comment:
most of this file is copied from `generator-executor.rs` but with the generator-specific parts replaced with PluginWorkProcessor